### PR TITLE
[Release Test] Add GCE variation for core release tests [4/n]

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
- dask_on_ray_100gb_sort
- stress_test_state_api_scale
- stress_test_many_tasks
- stress_test_dead_actors
- threaded_actors_stress_test
- many_nodes_actor_test_on_v2
- placement_group_performance_test
- scheduling_test_many_0s_tasks_many_nodes
- agent_stress_test

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_tests.yaml [hunk 1]
```diff
@@ -3891,13 +3891,22 @@
...
+        cluster_compute: dask_on_ray/dask_on_ray_sort_compute_template_gce.yaml
+
 
 - name: dask_on_ray_large_scale_test_spilling
   group: core-daily-test
   working_dir: nightly_tests
 
   frequency: nightly
-  team: core
+  team: data
```
Comment (can-anyscale) [inline release/release_tests.yaml:3909]: do you mean to change this

**Review by can-anyscale — APPROVED** [review]

File: release/release_tests.yaml (hunk 1 above)
Comment (jjyao) [reply release/release_tests.yaml:3909]: Yes, the chaos version of this test is already re-assigned to the data team.

**Comment by jjyao** [comment]
All the tests are passing, expect `many_nodes_actor_test_on_v2`: https://buildkite.com/ray-project/release-tests-pr/builds/35131

**Comment by can-anyscale** [comment]
GYI, the many_actor seems to fail on aws as well: https://buildkite.com/ray-project/release-tests-branch/builds/1552#01877785-50ac-4cc9-ab95-1f34dff1735b

**Pull request merged**